versiondb/tsrocksdb/comparator.go

```diff
@@ -0,0 +1,52 @@
+package tsrocksdb
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/linxGnu/grocksdb"
+)
+
+// CreateTSComparator should be compatible with builtin timestamp comparator.
+func CreateTSComparator() *grocksdb.Comparator {
+	return grocksdb.NewComparatorWithTimestamp(
+		"leveldb.BytewiseComparator.u64ts", TimestampSize, compare, compareTS, compareWithoutTS,
+	)
+}
+
+// compareTS compares timestamp as little endian encoded integers
+func compareTS(bz1 []byte, bz2 []byte) int {
+	ts1 := binary.LittleEndian.Uint64(bz1)
+	ts2 := binary.LittleEndian.Uint64(bz2)
+	switch {
+	case ts1 < ts2:
+		return -1
+	case ts1 > ts2:
+		return 1
+	default:
+		return 0
+	}
+}
+
+// compare compares two internal keys with timestamp surfix, larger timestamp comes first.
+func compare(a []byte, b []byte) int {
+	ret := compareWithoutTS(a, true, b, true)
```

Comment:
do we have other use cases for compareWithoutTS? If no, we can simplify the input to only a & b?